engine/engine_test.go

```diff
@@ -3340,6 +3340,49 @@ func TestNativeHistograms(t *testing.T) {
 	t.Run("float_histograms", func(t *testing.T) {
 		testNativeHistograms(t, cases, opts, generateFloatHistogramSeries)
 	})
+	t.Run("mixed_histograms", func(t *testing.T) {
+		testMixedNativeHistogramTypes(t, opts)
+	})
+}
+
+func testMixedNativeHistogramTypes(t *testing.T, opts promql.EngineOpts) {
+	test, err := promql.NewTest(t, "")
+	testutil.Ok(t, err)
+	defer test.Close()
+
+	var samples []tsdbutil.Sample
+
+	histograms := tsdbutil.GenerateTestHistograms(2)
+
+	for i, th := range histograms {
```

Comment:
Let's use explicit indexes to access histograms. The `for` statement obfuscates what the code tries to do.